File: python/7.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """

        if x > 0 :
            s = str(x)
            ans = int(s[::-1])
            if ans > 2**31-1:
                return 0
            else:
                return ans
        if x == 0:
            return 0
        else:
            x = abs(x)
            s = str(x)
            ans = int(s[::-1])
            if ans > 2**31:
                return 0
            return -ans

File: java/7.java
class Solution {
    public int reverse(int x) {
        int z = x < 0 ? -1 : 1;
        // x = Math.abs(x);
        long xx  = Math.abs((long)x);
        long y = 0;
        while(xx != 0){
            y = y * 10 + xx % 10;
            xx = xx / 10;
        }
        if(y > 2147483647){
            return 0;
        }
        return z *(int)y;
        
    }
}